Mathlib/CategoryTheory/Adjunction/Mates.lean

```diff
@@ -253,7 +253,7 @@ instance transferNatTransSelf_symm_iso (f : R₁ ⟶ R₂) [IsIso f] :
 then `f` is an isomorphism.
 The converse is given in `transferNatTransSelf_iso`.
 -/
-theorem transferNatTransSelf_of_iso (f : L₂ ⟶ L₁) [IsIso (transferNatTransSelf adj₁ adj₂ f)] :
+theorem transferNatTransSelf_of_iso (f : L₂ ⟶ L₁) [i : IsIso (transferNatTransSelf adj₁ adj₂ f)] :
```

Comment:
I'm not fully convinced by the "Let's name an instance argument only because I want to mention it as a named argument later". Instance arguments should usually not be named, and quite definitely not be fed in explicitly.

Can you fix the call site in a way that doesn't involve feeding the instance explicitly, and revert the naming of this instance argument?

Reply:
I think the named typeclass arguments are here to be used within the declaration itself, in place of an unwiedly long `‹type of the instance assumption›` (although I do happen to prefer the latter over the former. What you're doing in contrast is to refer to the name *outside* of the declaration, which is more brittle and not something I've seen anywhere else.

Reply:
I see. Anyway, those changes have been reverted. Thanks for looking at this. I won't add any more changes, so it's ready to be reviewed.